litmus-portal/cluster-agents/subscriber/pkg/k8s/operations.go

```diff
@@ -157,53 +159,48 @@ func IsClusterConfirmed() (bool, string, error) {
 }
 
 // ClusterRegister function creates litmus-portal config map in the litmus namespace
-func ClusterRegister(clusterData map[string]string) (bool, error) {
-	clientset, err := GetGenericK8sClient()
+func ClusterRegister(accessKey string) (bool, error) {
+	clientset, err:= GetGenericK8sClient()
 	if err != nil {
 		return false, err
 	}
 
-	newConfigMapData := map[string]string{
-		"IS_CLUSTER_CONFIRMED": clusterData["IS_CLUSTER_CONFIRMED"],
-		"SERVER_ADDR":          clusterData["SERVER_ADDR"],
-		"AGENT_SCOPE":          clusterData["AGENT_SCOPE"],
-		"COMPONENTS":           clusterData["COMPONENTS"],
-		"START_TIME":           clusterData["START_TIME"],
-		"VERSION":              clusterData["VERSION"],
-		"SKIP_SSL_VERIFY":      clusterData["SKIP_SSL_VERIFY"],
-		"CUSTOM_TLS_CERT":      clusterData["CUSTOM_TLS_CERT"],
+	isClusterConfirmed, _, err := IsClusterConfirmed()
+	if err != nil {
+		return false, err
 	}
 
-	_, err = clientset.CoreV1().ConfigMaps(AgentNamespace).Update(context.TODO(), &corev1.ConfigMap{
-		Data: newConfigMapData,
-		ObjectMeta: metav1.ObjectMeta{
-			Name: AgentConfigName,
-		},
-	}, metav1.UpdateOptions{})
+	configMapPatch:= map[string]bool {
+		"data": isClusterConfirmed,
```

Comment:
Did you test if this works? You need to specify the key-value mapping for the patch, here you just put the bool value directly under `data`

Reply:
@achiverram28  You can try something like this
		
            updatedConfigMap := corev1.ConfigMap{
			Data: map[string]string{
				"<key>": "<value>",
			},
		}

		configMapPatch, err := json.Marshal(updatedConfigMap)
		if err != nil {
			return fmt.Errorf("failed to marshal patched config, err: %v", err)
		}